lib/private/legacy/app.php

```diff
@@ -158,8 +158,9 @@ public static function loadApp(string $app) {
 					throw $ex;
 				}
 				\OC::$server->getLogger()->logException($ex);
-				if (!\OC::$server->getAppManager()->isShipped($app)) {
-					// Only disable apps which are not shipped
+
+				if (!\OC::$server->getAppManager()->isShipped($app) && !(isset(self::$appTypes[$app]) && self::$appTypes[$app] === 'authentication')) {
```

Comment:
Apps can have multiple types?

Reply:
Yeah no idea. What I saw not so far.

Reply:
They can and types is an array. But there is a method to help you:

```suggestion
				if (!\OC::$server->getAppManager()->isShipped($app) && !self::isType($app, 'authentication')) {
```